consul/structs/structs.go

```diff
@@ -137,6 +137,7 @@ type RegisterRequest struct {
 	Address    string
 	Service    *NodeService
 	Check      *HealthCheck
+	Checks     HealthChecks
```

Comment:
I don't think the FSM / State Store handle this, so I think you need to make changes to the RegisterRequest path too